cli/azd/pkg/tools/kubectl/util.go

```diff
@@ -0,0 +1,148 @@
+package kubectl
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/sethvargo/go-retry"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	ErrResourceNotFound = errors.New("cannot find resource")
+	ErrResourceNotReady = errors.New("resource is not ready")
+)
+
+func GetResource[T any](
+	ctx context.Context,
+	cli KubectlCli,
+	resourceType ResourceType,
+	resourceName string,
+	flags *KubeCliFlags,
+) (T, error) {
+	if flags == nil {
+		flags = &KubeCliFlags{}
+	}
+
+	if flags.Output == "" {
+		flags.Output = "json"
+	}
+
+	var zero T
+
+	res, err := cli.Exec(ctx, flags, "get", string(resourceType), resourceName)
+	if err != nil {
+		return zero, fmt.Errorf("failed getting resources, %w", err)
+	}
+
+	var resource T
+
+	switch flags.Output {
+	case "json":
+		err = json.Unmarshal([]byte(res.Stdout), &resource)
+		if err != nil {
+			return zero, fmt.Errorf("failed unmarshalling resources JSON, %w", err)
+		}
+	case "yaml":
+		err = yaml.Unmarshal([]byte(res.Stdout), &resource)
+		if err != nil {
+			return zero, fmt.Errorf("failed unmarshalling resources YAML, %w", err)
+		}
+	default:
+		return zero, fmt.Errorf("failed unmarshalling resources. Output format '%s' is not supported", flags.Output)
```

Comment:
should this be a panic?  
What would I do as a customer if I see this error? 
This is something which should never ever happen right?

Reply:
Going to stick with the standard error for the time being.  Correct, it should not happen unless an incorrect output format was specified.